src/lib/storage/segment_encoding_utils.cpp

```diff
@@ -40,16 +39,34 @@ std::unique_ptr<BaseSegmentEncoder> create_encoder(EncodingType encoding_type) {
   return encoder->create_new();
 }
 
-std::shared_ptr<BaseEncodedSegment> encode_segment(EncodingType encoding_type, DataType data_type,
-                                                   const std::shared_ptr<const BaseValueSegment>& segment,
-                                                   std::optional<VectorCompressionType> zero_suppression_type) {
-  auto encoder = create_encoder(encoding_type);
+std::shared_ptr<BaseEncodedSegment> encode_and_compress_segment(const std::shared_ptr<const BaseSegment>& segment,
+                                                                const DataType data_type,
+                                                                const SegmentEncodingSpec& encoding_spec) {
+  auto encoder = create_encoder(encoding_spec.encoding_type);
 
-  if (zero_suppression_type) {
-    encoder->set_vector_compression(*zero_suppression_type);
+  if (encoder->uses_vector_compression() && encoding_spec.vector_compression_type) {
```

Comment:
In which case would one of these be true and the other false?

Reply:
To be honest, I thought the check is not necessary, but valgrind complained and I was not able to reproduce it locally. The check fixed it for the CI setup. I will take another look at it.

Reply:
I'd be interested in the valgrind message

Reply:
Cannot reproduce. `encoder->uses_vector_compression()` has been removed again.